### PR TITLE
iptables: move to default recipe

### PR DESCRIPTION
**Pull request by hilmarm**
Based on xtables lock MR, but not necessary to have it.

First remove our custom file copying scheme entirely, then readd the bbappend file with a single patch that creates our wanted iptables and ip6tables rule files

### CODE REVIEW

**Comment by hilmarm** [comment]
Two questions,
1: should i drop the first commit with the iptables lock fix, since this is already in the default dunfell recipe?
2: Should i rename the patch prefix to something other than 0001, since the default dunfell recipe already contains three patches, and the prefixes 0001 and 0002 are already used there. In that case, what should I rename the prefix to?

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
This PR needs a proper title.

Also just remove the parts that copy/install our own service files, and leave the rest intact, I don't think we need to patch those files.

**Comment by hilmarm** [comment]
> This PR needs a proper title.
> 
> Also just remove the parts that copy/install our own service files, and leave the rest intact, I don't think we need to patch those files.

Edited commits to only remove our custom file additions, then readded the needed patch and entries in bbappend file

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
No, don't remove the rules files, just keep them. No patches. Just a bbappend that replaces the (empty) rules files.

**Comment by hilmarm** [comment]
Image built here: http://repo.bisdn.de.s3-eu-central-1.amazonaws.com/latest/accton-as4610/master/onie-bisdn-full-accton-as4610-202111031131.bin

Rebooting older images will quickly fail, as either iptables or ip6tables will fail to start

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
In general looks okay, but drop the first commit, no point in modifying a file we'll delete right after that.

**Comment by hilmarm** [comment]
> In general looks okay, but drop the first commit, no point in modifying a file we'll delete right after that.

First commit dropped entirely

File: recipes-extended/iptables/iptables_%.bbappend [hunk 1]
```diff
@@ -2,9 +2,7 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 
 SRC_URI += " \
     file://ip6tables.rules \
-    file://ip6tables.service \
     file://iptables.rules \
-    file://iptables.service \
 "
 
 inherit systemd
```
Comment (KanjiMonster) [inline recipes-extended/iptables/iptables_%.bbappend:8]: You probably won't need this one anymore ...

File: recipes-extended/iptables/iptables_%.bbappend [hunk 2]
```diff
@@ -14,14 +12,6 @@ do_install_append() {
...
-    install -d ${D}${systemd_system_unitdir}
-    install -m 0644 ${WORKDIR}/iptables.service \
-                    ${WORKDIR}/ip6tables.service \
-                    ${D}${systemd_system_unitdir}
-
-    sed -i -e 's,@SBINDIR@,${sbindir},g' ${D}${systemd_system_unitdir}/iptables.service \
-                                         ${D}${systemd_system_unitdir}/ip6tables.service
 }
 
 SYSTEMD_SERVICE_${PN} = "iptables.service ip6tables.service"
```
Comment (KanjiMonster) [inline recipes-extended/iptables/iptables_%.bbappend:17]: since you shouldn't need this anymore.

**Review by KanjiMonster — CHANGES_REQUESTED** [review]

**Comment by hilmarm** [comment]
> Have you tested the changes?
> 
> Also please squash fhe commits and drop the (fix) from the commit subject, it should be just ": "

No I am still building the image, commits will be squashed once I finish testing new image

**Comment by hilmarm** [comment]
Image built and tested with latest changes, ready for re-review.

Test image can be found here:
http://repo.bisdn.de.s3-eu-central-1.amazonaws.com/latest/agema-ag7648/master/onie-bisdn-full-agema-ag7648-202111031604.bin

File: recipes-extended/iptables/iptables_%.bbappend (hunk 2 above)
Comment (hilmarm) [reply recipes-extended/iptables/iptables_%.bbappend]: also removed

**Comment by KanjiMonster** [comment]
FWIW, this has been added/fixed in yocto itself since dunfell, with commit http://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?h=dunfell&id=7a2c2df853934a77a154d254b5d932b9817283f8 (which added the IPv6 helper).

**Review by KanjiMonster — APPROVED** [review]

**Pull request merged**